staging/src/k8s.io/apimachinery/pkg/apimachinery/announced/group_factory.go

```diff
@@ -89,6 +89,17 @@ func (gmf *GroupMetaFactory) Announce(groupFactoryRegistry APIGroupFactoryRegist
 		panic(err)
 	}
 	return gmf
+
+}
+
+// AnnounceIfNotAnnounced is the same as Announce, except that it returns nil
```

Comment:
Eliminating double registration is a much bigger refactor than i expected. I had refactored client-go to not relying on api.Scheme/Registry/Codecs (https://github.com/kubernetes/kubernetes/pull/44523), but other packages like kubectl, testapi, federation and more are still relying on api.XXX, and they rely on multiple registration to make sure the API are registered.